rig-core/src/providers/gemini/completion.rs

```diff
@@ -0,0 +1,798 @@
+// ================================================================
+//! Google Gemini Completion Integration
+//! From [Gemini API Reference](https://ai.google.dev/api/generate-content)
+// ================================================================
+
+/// `gemini-1.5-flash` completion model
+pub const GEMINI_1_5_FLASH: &str = "gemini-1.5-flash";
+/// `gemini-1.5-pro` completion model
+pub const GEMINI_1_5_PRO: &str = "gemini-1.5-pro";
+/// `gemini-1.5-pro-8b` completion model
+pub const GEMINI_1_5_PRO_8B: &str = "gemini-1.5-pro-8b";
+/// `gemini-1.0-pro` completion model
+pub const GEMINI_1_0_PRO: &str = "gemini-1.0-pro";
+
+use gemini_api_types::{
+    Content, ContentCandidate, FunctionDeclaration, GenerateContentRequest,
+    GenerateContentResponse, GenerationConfig, Part, Role, Tool,
+};
+use std::convert::TryFrom;
+
+use crate::completion::{self, CompletionError, CompletionRequest};
+
+use super::Client;
+
+// =================================================================
+// Rig Implementation Types
+// =================================================================
+
+#[derive(Clone)]
+pub struct CompletionModel {
+    client: Client,
+    pub model: String,
+}
+
+impl CompletionModel {
+    pub fn new(client: Client, model: &str) -> Self {
+        Self {
+            client,
+            model: model.to_string(),
+        }
+    }
+}
+
+impl completion::CompletionModel for CompletionModel {
+    type Response = GenerateContentResponse;
+
+    async fn completion(
+        &self,
+        mut completion_request: CompletionRequest,
+    ) -> Result<completion::CompletionResponse<GenerateContentResponse>, CompletionError> {
+        let mut full_history = Vec::new();
+        full_history.append(&mut completion_request.chat_history);
+
+        let prompt_with_context = completion_request.prompt_with_context();
+
+        full_history.push(completion::Message {
+            role: "user".into(),
+            content: prompt_with_context,
+        });
+
+        // Handle Gemini specific parameters
+        let mut generation_config =
+            GenerationConfig::try_from(completion_request.additional_params.unwrap_or_default())?;
```

Comment:
This like here causes requests to fail if the user does not set `additional_params`. 

FYI the default value for `serde_json::Value` is `serde_json::Value::Null` and not an empty `serde_json::Value::Object`, so if the user does not set `additional_params`, this code will set it to `null` before calling `GenerationConfig::try_from` which will fail.

Two possible fixes:
1. Use `unwrap_or(serde_json::Value::Object({}))` instead of `unwrap_or_default()`
2. Only call `GenerationConfig::try_from` if `additional_params` is not `None`